src/core/abuse-core.js

```diff
@@ -0,0 +1,196 @@
+              });
+          });
+      }
+    });
+}
+
+/*
+  Resolves the given feed item report, whether it is banned or not.
+  If the item is banned, it also resolves all of the other reports
+  for the same item. As of this writing not banning does not resolve
+  all of the reports, could be changed quite easily.
+*/
+function resolveReport(reportParams) {
+  return knex.transaction(function(trx) {
+    return trx("feed_item_reports")
+      .update("is_resolved", true)
+      .where("id", reportParams.reportId)
+      .returning(["id", "feed_item_id"])
+      .then(reports => {
+        if (_.isEmpty(reports)) {
+          throw new Error("Feed item report resolving failed: " + reportParams);
+        }
+        if (reportParams.banned) {
+          return trx("feed_items")
+            .update("is_banned", true)
+            .where("id", reports[0].feed_item_id)
+            .returning("id")
+            .then(feedItems => {
+              if (_.isEmpty(feedItems)) {
+                throw new Error(
+                  "Feed item not found: " + reports[0].feed_item_id
+                );
+              }
+              return trx("feed_item_reports")
+                .update("is_resolved", true)
+                .where("feed_item_id", feedItems[0])
+                .returning("id");
+            });
+        }
+      })
+      .catch(err => {
+        throw err;
+      });
+  });
+}
+
+/*
+  Fetches all of the reported feed items after a certain date.
+  Also fetches information on the reporter and the feed item itself.
+*/
+function getReportedFeedItems(params) {
+  let whereClause = [];
+  if (params.beforeId) {
+    whereClause.push("feed_item_reports.id < " + params.beforeId + " AND ");
+  }
+  whereClause += "is_resolved = false";
+
+  return knex("feed_item_reports")
+    .count("id")
+    .where("is_resolved", false)
+    .then(number_of_rows => {
+      return knex
+        .from("feed_item_reports")
+        .select(
+          "feed_item_reports.id as report_id",
+          "feed_item_reports.feed_item_id as id",
+          "report_creator_id",
+          "feed_item_reports.created_at as createdAt",
+          "report_description",
+          "is_resolved",
+          "user_id",
+          "location",
+          "feed_items.image_path",
+          "teams.name as team_name",
+          "users.name",
+          "users.id as user_id",
+          "text",
+          "type",
+          "feed_items.is_banned",
+          "hot_score",
+          "teams.city_id",
+          "parent_id"
+        )
+        .innerJoin(
+          "feed_items",
+          "feed_item_reports.feed_item_id",
+          "feed_items.id"
+        )
+        .innerJoin("users", "feed_item_reports.report_creator_id", "users.uuid")
+        .innerJoin("teams", "users.team_id", "teams.id")
+        .whereRaw(whereClause)
+        .limit(20)
+        .orderBy("report_id", "desc")
+
+        .then(feed => {
+          for (var i = 0; i < feed.length; i++) {
```

Comment:
I would maybe rewrite this, defining the imgix image url logic into its own function
```
const maybeImgixUrl = (feedItem) => {
  if (feedItem.type === "IMAGE") {
    const { image_path } = feedItem;

    if (process.env.DISABLE_IMGIX === "true" || _.endsWith(imagePath, "gif")) {
      return `${GCS_CONFIG.baseUrl}/${GCS_CONFIG.bucketName}/${image_path}`;
    } else {
      return `https://${GCS_CONFIG.bucketName}.imgix.net/${image_path}${process.env.IMGIX_QUERY}`;
    }
  }

  return feedItem.url;
}
```

and then loop through the elements using Array.map

```
.then(feedItems => feedItems.map(feedItem =>
  Object.assign({}, feedItem, {
    author: {
      id: feedItem.user_id,
      name: feedItem.name,
      team: feedItem.team_name,
    },
    url: maybeImgixUrl(feedItem)
  })
));
```

When writing for loops in javascript, it is a good practice to always use block scoped `let` instead of `var` for the iterator variable.